File: Problem Solving/Algorithms/Strings/Game of Thrones - I.cpp
#include <bits/stdc++.h>
#include <map>

using namespace std;

/*
 * Complete the 'gameOfThrones' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

string gameOfThrones(string s) {
    map<char, int> freq;
    for (char ch: s) {
        if (freq.find(ch) == freq.end())
            freq[ch] = 1;
        else
            freq[ch]++;
    }

    int odds = 0;
    for (auto p: freq)
        if (p.second % 2 == 1)
            odds++;

    if (s.size() % 2 == 1) {
        if (odds <= 1)
            return "YES";
        return "NO";
    } else {
        if (odds > 0)
            return "NO";
        return "YES";
    }

}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = gameOfThrones(s);

    fout << result << "\n";

    fout.close();

    return 0;
}

File: Problem Solving/Algorithms/Strings/Game of Thrones - I.py
#!/bin/python3

import math
import os
import random
import re
import sys
from collections import defaultdict

#
# Complete the 'gameOfThrones' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#


def gameOfThrones(s):
    # Write your code here
    freq = defaultdict(int)
    for item in s:
        freq[item] += 1

    odds = 0
    for v in freq.values():
        if v % 2 == 1:
            odds += 1

    if len(s) % 2 == 1:
        if odds <= 1:
            return "YES"
        return "NO"
    else:
        if odds > 0:
            return "NO"
        return "YES"


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = gameOfThrones(s)

    fptr.write(result + '\n')

    fptr.close()
